lib/collection/src/update_handler.rs

```diff
@@ -352,4 +344,34 @@ impl UpdateHandler {
             .send(OptimizerSignal::Stop)
             .unwrap_or_else(|_| debug!("Optimizer already stopped"));
     }
+
+    async fn flush_worker(
+        segments: LockedSegmentHolder,
+        wal: Arc<Mutex<SerdeWal<CollectionUpdateOperations>>>,
+        flush_interval_sec: u64,
+        mut stop_receiver: oneshot::Receiver<()>,
+    ) {
+        loop {
+            // Stop flush worker on signal or if sender was dropped
+            // Even if timer did not finish
+            tokio::select! {
+                _ = tokio::time::sleep(Duration::from_secs(flush_interval_sec)) => {},
+                _ = &mut stop_receiver => {
+                    debug!("Stopping flush worker.");
+                    return;
+                }
+            };
+
+            trace!("Attempting flushing");
+            let confirmed_version = {
+                let read_segments = segments.read();
+                let flushed_version = read_segments.flush_all().unwrap();
```

Comment:
Can we report this error similar to https://github.com/qdrant/qdrant/blob/e45379e4384062e92ee1c9be82c250047464c9ef/lib/collection/src/update_handler.rs#L201 ? We can probably rename `optimizer_errors` -> `persistance_errors` or something.

Reply:
Fixed, though left the same naming. Can't think of a better name for both flush and optimization, I guess like previously we can consider flushing as part of the optimization.